docs/mmds/mmds-user-guide.md

```diff
@@ -0,0 +1,250 @@
+# microVM Metadata Service
+
+The Firecracker microVM Metadata Service (MMDS) is a mutable data store which
+can be used for sharing information between host and guests, in a secure and
+easy at hand way.
+
+# Activating the microVM Metadata Service
+
+By default, MMDS is not reachable from the guest operating system. At microVM
+runtime, MMDS is tightly coupled with a network interface, which allows MMDS
+requests. When configuring the microVM, if MMDS needs to be activated, a
+network interface has to be configured to allow MMDS requests. Network
+interface configuration API can be found in the
+[firecracker swagger file](../../src/api_server/swagger/firecracker.yaml).
+
+### Example
+
+```bash
+MMDS_NET_IF=eth0
+curl --unix-socket /tmp/firecracker.socket -i \
+  -X PUT 'http://localhost/network-interfaces/${MMDS_NET_IF}' \
+  -H 'Accept: application/json' \
+  -H 'Content-Type: application/json' \
+  -d '{
+      "iface_id": "${MMDS_NET_IF}",
+      "guest_mac": "AA:FC:00:00:00:01",
+      "host_dev_name": "tap0"
+      "allow_mmds_requests": true
+    }'
+```
+
+# Configuring the microVM Metadata Service
+
+MMDS can be configured pre-boot only, using the Firecracker API server. This
+can be achieved through a HTTP `PUT` request to `/mmds/config` resource. The
+complete MMDS configuration API is described in the [firecracker swagger 
+file](../../src/api_server/swagger/firecracker.yaml).
+
+At the moment, MMDS is configurable with respect to the IPv4 address used by
+guest applications when issuing requests to MMDS. If MMDS configuration is not
+provided before booting up the guest, the MMDS IPv4 address defaults to
+`169.254.169.254`.
+ 
+### Example
+
+```bash
+MMDS_IPV4_ADDR=169.254.170.2
+curl --unix-socket /tmp/firecracker.socket -i \
+    -X PUT "http://localhost/mmds/config" \
+    -H "accept: application/json" \
+    -H "Content-Type: application/json" \
+    -d "{
+             "ipv4_address": "${MMDS_IPV4_ADDR}"      
```

Comment:
Please keep alignment consistent - I prefer the alignment at line 24 which wastes less space

Reply:
This applies to all `curl` invocation examples